### PR TITLE
Am 1087 collate db queries

### PR DESCRIPTION
**Pull request by nitishsingla1234**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/AM-1087

### Change description ###
Collate DB queries.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/roleassignment/domain/service/deleteroles/DeleteRoleAssignmentOrchestrator.java [hunk 1]
```diff
@@ -34,16 +35,19 @@
...
+    private PersistenceUtil persistenceUtil;
     RequestEntity requestEntity;
     AssignmentRequest assignmentRequest;
     Request request;
 
     public DeleteRoleAssignmentOrchestrator(PersistenceService persistenceService,
                                             ParseRequestService parseRequestService,
-                                            ValidationModelService validationModelService) {
+                                            ValidationModelService validationModelService,
+                                            PersistenceUtil persistenceUtil) {
```
Comment (dharmendra8186) [inline src/main/java/uk/gov/hmcts/reform/roleassignment/domain/service/deleteroles/DeleteRoleAssignmentOrchestrator.java:46]: why we need persistenceUtil?

File: src/main/java/uk/gov/hmcts/reform/roleassignment/util/PersistenceUtil.java [hunk 2]
```diff
@@ -150,4 +152,21 @@ public RoleAssignment convertEntityToRoleAssignment(RoleAssignmentEntity roleAss
                                 Collections.emptyList())
             .build();
     }
+
+    public HistoryEntity prepareHistoryEntityForPersistance(RoleAssignment roleAssignment, Request request) {
+        UUID roleAssignmentId = roleAssignment.getId();
+        UUID requestId = request.getId();
```
Comment (dharmendra8186) [inline src/main/java/uk/gov/hmcts/reform/roleassignment/util/PersistenceUtil.java:158]: Why we moved this method from Persistence Service to here?

**Review by dharmendra8186 — COMMENTED** [review]
Please check review comments.

File: src/main/java/uk/gov/hmcts/reform/roleassignment/util/PersistenceUtil.java (hunk 2 above)
Comment (nitishsingla1234) [reply src/main/java/uk/gov/hmcts/reform/roleassignment/util/PersistenceUtil.java:158]: Hi Dharmendra,
Earlier, the persistance was happening through persistenceService.addHistoryEntity() method. Now, we have removed the persistance logic from old method and moved it outside loop. Current method "prepareHistoryEntityForPersistance" is only preparing the object and not doing any persist operation. Hence moved it from PersistabceService to PersistanceUtil.

File: src/main/java/uk/gov/hmcts/reform/roleassignment/domain/service/deleteroles/DeleteRoleAssignmentOrchestrator.java (hunk 1 above)
Comment (nitishsingla1234) [reply src/main/java/uk/gov/hmcts/reform/roleassignment/domain/service/deleteroles/DeleteRoleAssignmentOrchestrator.java:46]: Hi Dharmendra,
Please see the other comment.
Thanks

File: src/main/java/uk/gov/hmcts/reform/roleassignment/util/PersistenceUtil.java (hunk 2 above)
Comment (dharmendra8186) [reply src/main/java/uk/gov/hmcts/reform/roleassignment/util/PersistenceUtil.java:158]: Ok Fine.

**Review by aashishaggarwal123 — APPROVED** [review]

**Pull request merged**